ocs_ci/ocs/ui/base_ui.py

```diff
@@ -195,6 +202,86 @@ def take_screenshot(self):
+
+        Args:
+            locator (tuple): (GUI element needs to operate on (str), type (By))
+            timeout (int): Looks for a web element until timeout (sec) occurs
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((locator[1], locator[0])))
+        element.clear()
+
+    def wait_for_element(self, locator, timeout=30):
+        """
+        Method to wait for a web element to be found (use of explicit wait type).
+        The method ignored all the listed exceptions for the given timeout and waits for the element
+        to be visible on the UI.
+
+        Args:
+            locator (tuple): (GUI element needs to operate on (str), type (By))
+            timeout (int): Looks for a web element repeatedly until timeout (sec) occurs
+        return:
+            bool: True if the element is found, False otherwise
+
+        """
+        wait = WebDriverWait(
+            self.driver,
+            timeout=timeout,
+            poll_frequency=1,
+            ignored_exceptions=[
+                NoSuchElementException,
+                ElementNotVisibleException,
+                ElementNotSelectableException,
+                TimeoutException,
+            ],
+        )
+        try:
+            wait.until(ec.visibility_of_element_located(locator[::-1]))
+            return True
+        except TimeoutException:
+            logger.error("Could not find expected element")
+            return False
+
+    def wait_until_expected_text_is_found(self, locator, expected_text, timeout=300):
+        """
+        Method to wait for a expected text to appear on the UI (use of explicit wait type),
+        this method is helpful in working with elements which appear on completion of certain UI action and
+        ignores all the listed exceptions for the given timeout.
+
+        Args:
+            locator (tuple): (GUI element needs to operate on (str), type (By))
+            expected_text (str): Text which needs to be searched on UI
+            timeout (int): Looks for a web element repeatedly until timeout (sec) occurs
+        return:
+            str: Returns the text (string) when found
+
+        """
+        wait = WebDriverWait(
+            self.driver,
+            timeout=timeout,
+            poll_frequency=1,
+            ignored_exceptions=[
+                NoSuchElementException,
+                ElementNotVisibleException,
+                ElementNotSelectableException,
+                TimeoutException,
+            ],
+        )
+        try:
+            wait.until(
+                ec.text_to_be_present_in_element(
+                    (locator[1], locator[0]), text_=expected_text
+                )
+            )
+            logger.info(f"Element text found: {expected_text} ")
+            return f"{expected_text}"
+        except TimeoutException:
+            logger.error(f"Could not find expected text: {expected_text} ")
```

Comment:
what should be returned if the exception is caught?

Reply:
Logger error will be raised along with TimeoutException. In my case, since I am using it in Assertion, so Assertion will also fail and execution will stop with AssertionError.

Reply:
The return value will be *None* if TimeoutException is occurred.

Reply:
Thanks @ebenahar for pointing it out, please review the modifications.

Reply:
thanks, I see it's changed and looks good to me

Reply:
@ebenahar minor code change, please review